static/app/components/forms/fieldFromConfig.tsx

```diff
@@ -1,105 +1,101 @@
-import {Component} from 'react';
+import * as React from 'react';
 
 import {Scope} from 'sentry/types';
 
 import BlankField from './blankField';
-import BooleanField from './booleanField';
-import ChoiceMapperField from './choiceMapperField';
-import EmailField from './emailField';
+import BooleanField, {BooleanFieldProps} from './booleanField';
+import ChoiceMapperField, {ChoiceMapperFieldProps} from './choiceMapperField';
+import EmailField, {EmailFieldProps} from './emailField';
+import {FieldProps} from './field';
 import FieldSeparator from './fieldSeparator';
-import HiddenField from './hiddenField';
-import InputField from './inputField';
+import HiddenField, {HiddenFieldProps} from './hiddenField';
+import InputField, {InputFieldProps} from './inputField';
 import NumberField from './numberField';
 import ProjectMapperField from './projectMapperField';
-import RadioField from './radioField';
-import RangeField from './rangeField';
-import SelectAsyncField from './selectAsyncField';
-import SelectField from './selectField';
-import SentryProjectSelectorField from './sentryProjectSelectorField';
+import RadioField, {RadioFieldProps} from './radioField';
+import RangeField, {RangeFieldProps} from './rangeField';
+import SelectAsyncField, {SelectAsyncFieldProps} from './selectAsyncField';
+import SelectField, {SelectFieldProps} from './selectField';
+import SentryProjectSelectorField, {RenderFieldProps} from './sentryProjectSelectorField';
 import TableField from './tableField';
-import TextareaField from './textareaField';
+import TextareaField, {TextareaFieldProps} from './textareaField';
 import TextField from './textField';
 import {Field} from './type';
 
-type Props = {
+interface FieldFromConfigProps {
   field: Field;
   access?: Set<Scope>;
+
   disabled?: boolean | ((props) => boolean);
   flexibleControlStateSize?: boolean;
   getData?: (data) => any;
   highlighted?: boolean;
   inline?: boolean;
+  name?: string;
   noOptionsMessage?: () => string;
   onBlur?: (value, event) => void;
   stacked?: boolean;
-};
+}
 
-export default class FieldFromConfig extends Component<Props> {
-  render() {
-    const {field, ...otherProps} = this.props;
+function FieldFromConfig(props: FieldFromConfigProps): React.ReactElement | null {
+  const {field, ...otherProps} = props;
 
-    const props = {
-      ...otherProps,
-      ...field,
-    };
+  const componentProps = {
+    ...otherProps,
+    ...field,
+  };
 
-    switch (field.type) {
-      case 'separator':
-        return <FieldSeparator />;
-      case 'secret':
-        return <InputField {...props} type="password" />;
-      case 'range':
-        // TODO(ts) The switch on field.type is not resolving
-        // the Field union for this component. The union might be 'too big'.
-        return <RangeField {...(props as any)} />;
-      case 'blank':
-        return <BlankField {...props} />;
-      case 'bool':
-      case 'boolean':
-        return <BooleanField {...props} />;
```

Comment:
I really think most of this just worked because we were using & which resulted in a very wide type that just supported all possible props... Given the comment on #52, I actually doubt if ts checking here actually worked properly

Reply:
i wouldn't' mind a ts-expect-error in that case. I never expected the types in these files to be great

Reply:
Yeah, at this point that's true, but we could write the component in a different way where we wouldn't need to do that. It's just too painful to fix at this point...

Reply:
@scttcper I think I got it working. I'll poke the dev env tomorrow to see if I broke anything 😅